engine/orchestration/src/main/java/org/apache/cloudstack/engine/cloud/entity/api/VirtualMachineEntityImpl.java

```diff
@@ -269,4 +270,23 @@ public void disconnectFrom(NetworkEntity netowrk, short nicId) {
 
     }
 
+    @Override
+    public void setParamsToEntity(Map<VirtualMachineProfile.Param, Object> map) {
+        if (this.vmEntityVO != null) {
+            Map<String, String> details = this.vmEntityVO.getDetails();
+
+            if (details == null) {
```

Comment:
Better get and validate details when map is not empty.

Reply:
Changes done as suggested